pkg/cmd/run.go

```diff
@@ -128,10 +139,18 @@ func NewRunCommand(factory factory.Factory, streams genericclioptions.IOStreams)
 		},
 	}
 
+	// flags for existing usage
 	cmd.Flags().StringVarP(&o.container, "container", "c", o.container, "Specify the container")
-	cmd.Flags().BoolVarP(&o.attach, "attach", "a", o.attach, "Whether or not to attach to the trace program once it is created")
 	cmd.Flags().StringVarP(&o.eval, "eval", "e", o.eval, "Literal string to be evaluated as a bpftrace program")
-	cmd.Flags().StringVarP(&o.program, "filename", "f", o.program, "File containing a bpftrace program")
+	cmd.Flags().StringVarP(&o.filename, "filename", "f", o.filename, "File containing a bpftrace program")
+
+	// flags for new generic interface
+	cmd.Flags().StringVar(&o.tracer, "tracer", "bpftrace", "Tracing system to use")
```

Comment:
i'm not super familiar with cobra, is it possible to have the list of supported tracers here also? For what users can select from?

Reply:
I didn't update the documentation yet because there is no other tracer. Do you think it's more appropriate to do that when `bcc` support is added in follow up PR?

Reply:
I can modify the list here to just include `bpftrace` for now though. Leo had a comment around automatic validation of these things with `go-validator` so will look into that too.